### DISCUSSION TITLE
why  resolver ignore NamedExport which has source , keep ctxt as zero

### DISCUSSION THREAD

**Discussion by stormslowly** [swc-project/swc#8408] [General]
```
// case 1
let a= 1;
export {a as b}

// case 2
export {a as b } from "c"
```
in case 1, b has been marked under unresolved marks.
but 
in case 2 , both a and b's ctxt are all 0

https://github.com/swc-project/swc/blob/2113bb3e19131d68eb2a1c384e4ac58b2bc10f00/crates/swc_ecma_transforms_base/src/resolver/mod.rs#L1079-L1085

the code shows that,  when NamedExport with source, resolver will not mark its ctxt.

anyone know why ?

**Comment by stormslowly** [comment]
why i care about the `ctxt`,  because i have a requirement, for example
if `b` is a not defined variable,   a stmt will be inject into the module eg `var b = require("some-mock-b")`

i use  `ctxt.outer() == unresolved_mark`, to check if it defined or not, in case 1, b is treated as unresolved; so an un-expected require inject to code.

in case 2,  the ctxt is undermark root, so  inject not happened.

**Reply by kdy1** [reply]
It's not a thing that the `core` is responsible for. Write your own one.

**Reply by stormslowly** [reply]
yes, i has wrote a own implementation for injecting.  my question is why 

1. `export {a as b}`,  b is marked as unresolved 
2. `export {a as b } from "c"` all idents' ctxt are zero .

all these by design or by mistake?

**Comment by kdy1** [comment]
By design

2023년 12월 9일 (토) 오전 8:38, pshu ***@***.***>님이 작성:

**Reply by stormslowly** [reply]
could you plz give me some hint why design like this. 
current ctxt makes me a little confused  : (

thank you so much : P
